src/tools/wix/Binder.cs

```diff
@@ -3507,23 +3500,20 @@ private bool BindBundle(Output bundle, string bundleFile)
             string layoutDirectory = Path.GetDirectoryName(bundleFile);
 
             // Handle any payloads not explicitly in a container.
-            foreach (string payloadName in allPayloads.Keys)
+            IEnumerable<PayloadInfoRow> payloadsWithNoContainer = allPayloads.Values.Where(p => string.IsNullOrEmpty(p.Container));
+            foreach (PayloadInfoRow payload in payloadsWithNoContainer)
             {
-                if (!payloadsAddedToContainers.ContainsKey(payloadName))
+                if (PackagingType.Embedded == payload.Packaging)
                 {
-                    PayloadInfoRow payload = allPayloads[payloadName];
-                    if (PackagingType.Embedded == payload.Packaging)
```

Comment:
If I understand your explanation, can't you do this with in a couple lines by adding

```
if (!String.IsNullOrEmpty(payload.Container))
{
    continue;
}
else if (!payloadsAddedToContainers.ContainsKey(payloadName))
```

and not doing anything else? The less lines touched, the better.

Also, that targeted fix seems incomplete without further investigation. The fact that the payload isn't added to the `ContainerInfo`'s `Payloads` around line 3427 makes me wonder if other things are broken.

Reply:
The cab payloads are added to the container in ChainPackageInfo.ResolveMsiPackage() line 863.

I preferred this solution to make the code cleaner, as the only use of payloadsAddedToContainers was to detect which payloads belong to no container.
I think using Linq is more readable

Reply:
Thanks.

Agreed, your code is cleaner. I probably would still take this as is. But at this point of v3's lifecycle, I would argue that it's more important that the diffs are cleaner. This PR is +17/-27 with changes scattered through the file. My proposed change would be +5/-1 with all of the lines together. Of course, all of this doesn't really matter until this is fixed (or verified as already fixed) in v4 and we agree to take this fix in v3.14.

Reply:
Well... most diffs are due to "Remove and Sort Usings" in VS

Reply:
Remove and sort usings are fine in v4 but not in v3 at this point in time.

Please remove extraneous code changes so we can better review the actual changes you propose.